app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -12,16 +12,24 @@ def auth_methods_session
   end
 
   def handle_verification_for_authentication_context(result:, auth_method:, extra_analytics: nil)
+    mfa_selection_attempt_count(auth_method)
     analytics.multi_factor_auth(
       **result.to_h,
       multi_factor_auth_method: auth_method,
       enabled_mfa_methods_count: mfa_context.enabled_mfa_methods_count,
       new_device: new_device?,
       **extra_analytics.to_h,
+      mfa_attempts: user_session[:mfa_attempts],
+      pii_like_keypaths: [
+        [:mfa_attempts, :otp],
+        [:errors, :personal_key],
+        [:error_details, :personal_key],
```

Comment:
It was in response to an error I was getting. I can take another look for more clarity.

Reply:
It was in the same spec that was causing the session sensitive key error
./spec/features/legacy_passwords_spec.rb:61

1) legacy passwords signing in with an incorrect uak personal key digest does not grant access
     Failure/Error:
                     raise PiiDetected, <<~ERROR
                       track_event received pii key path: #{current_keypath.inspect}
                       event: #{event} (#{constant_name})
                       full event: #{attributes.inspect}
                       allowlisted keypaths: #{pii_like_keypaths.inspect}
                     ERROR
     
     FakeAnalytics::PiiDetected:
       track_event received pii key path: [:errors, :personal_key]
       event: Multi-Factor Authentication ()
       full event: {:success=>false, :errors=>{:personal_key=>["Incorrect personal key"]}, :error_details=>{:personal_key=>{:personal_key_incorrect=>true}}, :new_device=>true, :mfa_attempts=>{"personal_key"=>1}, :multi_factor_auth_method=>"personal-key", :enabled_mfa_methods_count=>1}
       allowlisted keypaths: [[:mfa_attempts, :otp], [:error_details, :personal_key]]

Reply:
I'm not able to reproduce that test failure locally in your branch when removing this `pii_like_keypaths` code. Also, `mfa_attempts.otp` would never exist in the hash structure since  you changed that in an earlier commit.